### PR TITLE
enable coretext support for harfbuzz

### PR DESCRIPTION
**Pull request by tallytalwar**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -263,8 +268,36 @@ set(HARFBUZZ_SRC_FILES
...
+        list(APPEND THIRD_PARTY_LIBS ${CORE_TEXT})
+    endif (CORE_TEXT)
+
+    # CORE_GRAPHICS needed for ios
+    find_library(CORE_GRAPHICS CoreGraphics)
+    if (CORE_GRAPHICS)
+        list(APPEND THIRD_PARTY_LIBS ${CORE_GRAPHICS})
+    endif (CORE_GRAPHICS)
+
+    mark_as_advanced(APPLICATION_SERVICES_FRAMEWORK)
```
Comment (karimnaaji) [inline CMakeLists.txt]: Is this necessary?

File: CMakeLists.txt [hunk 2]
```diff
@@ -228,6 +228,11 @@ endif()
 
 ## Harfbuzz ##
 ##############
+
+if (APPLE)
+    option(HB_HAVE_CORETEXT "Enable CoreText shaper backend on macOS" ON)
```
Comment (karimnaaji) [inline CMakeLists.txt]: We should have that option at the top with other options, also `APPLE` is only defined to be used from tangram-es, we should rather control that option being set in tangram-es based on the platform used e.g.:

https://github.com/tangrams/tangram-es/blob/master/core/deps/CMakeLists.txt#L54
```
if(APPLE)
  set(HB_HAVE_CORETEXT ON CACHE BOOL "Enable core text support")
endif()
```

Also we should document `HB_HAVE_CORETEXT` from the readme and rename it to `HARFBUZZ_HAVE_CORETEXT`.

Comment (tallytalwar) [reply CMakeLists.txt]: Moved the `option` above with other `option`s. Discussed the usage of `APPLE` on slack and `APPLE` is a cmake keyword and works aptly here.

File: CMakeLists.txt (hunk 1 above)
Comment (tallytalwar) [reply CMakeLists.txt]: removed.

**Pull request merged**